contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -946,7 +948,13 @@ resolve_varbinary_typmod_from_exp(Node *expr)
 			if (!con->constisnull)
 			{
 				bytea	   *source = (bytea *) con->constvalue;
-
+				actualSize = VARSIZE_ANY(source);
```

Comment:
Use VARSIZE_ANY_EXHDR to find actual length of the input and compare it against 8000 (define new macro just like VARCHAR_MAX)